include/seqan3/argument_parser/detail/version_check.hpp

```diff
@@ -424,12 +425,12 @@ class version_checker
         return "powershell.exe -NoLogo -NonInteractive -Command \"& {Invoke-WebRequest -erroraction 'silentlycontinue' -OutFile";
 #else  // Unix based platforms.
         if (!system("wget --version > /dev/null 2>&1"))
-            return "wget -q -O";
+            return "wget --timeout=10 --tries=1 -q -O";
         else if (!system("curl --version > /dev/null 2>&1"))
-            return "curl -o";
+            return "curl --connect-timeout 10 -o";
     #ifndef __linux  // ftp call does not work on linux
         else if (!system("which ftp > /dev/null 2>&1"))
-            return "ftp -Vo";
+            return "ftp -timeout 10 -Vo";
```

Comment:
the ftp command doesn't have `-timeout` on OpenBSD and it cannot be used to make HTTP request so this definitely does not work. 
I would recommend the following:

```cpp
#elseif defined(__OpenBSD__)
    return "ftp -w10 -Vo";
#elseif defined(__FreeBSD__)
    return "fetch --timeout=10 -o";
#else
    return "";
#endif
```

Reply:
(you don't need to check whether the ftp/fetch commands are available, they are guaranteed to be present on those systems.

Reply:
The `ifndef linux` block needs to be removed now.

Reply:
I can assume ftp is available on those systems? 
I will add a comment though to make it clear that the code is fine as is.